sdk/servicebus/service-bus/test/public/sessionsTests.spec.ts

```diff
@@ -462,23 +468,20 @@ describe("SessionReceiver - disconnects", function(): void {
           (receiver as any)["_context"].connection["_connection"].idle();
         },
         async processError(err) {
-          err.error.message && (err.error as ServiceBusError).code.should.equal("SessionLockLost");
+          should.equal(
```

Comment:
You'll have to try/catch around this  - processError will still eat your exception if it doesn't match.

Reply:
The would still fail in the case of "exception from your assert".. `errorIsThrownResolver` won't be called on line 476.. which means.. line 482 keeps waiting forever and the test would fail.

Reply:
So, not changing anything here?

Reply:
I was thinking you could just call errorIsThrownRejector() instead :)

Or:
errorIsThrownResolver(err);

And then do the assert in the code that was already awaiting on the promise:

const err = await errorIsThrown;
should.equal(blah)

Reply:
Good.